scripts/rotate-api-tokens.sh

```diff
@@ -130,10 +130,78 @@ function remove_old_callback_token() {
   echo "Done."
 }
 
+function change_ft_account_passwords() {
+  create_credentials_update_branch
+
+  echo "Changing Functional Test DMP account passwords for ${STAGE}."
+
+  export STAGE_LOWER
+
+  # find all values of fields named _email which have an equivalent _password variable
+  for ACCOUNT_EMAIL in $(./sops-wrapper -d jenkins-vars/jenkins.yaml|yq -crM '[[(.smoulder_test_variables, .smoke_test_variables, .functional_test_variables)[env.STAGE_LOWER]|values|to_entries]|flatten|.[]|select(.key|endswith("_email")).value]|unique|.[]') ; do
+    export ACCOUNT_EMAIL
+    # passwords for a particular email address need to be common across test variants becuase they
+    # share the same instance
+    new_password=$(generate_api_token)
+
+    for VARIANT in smoulder_test_variables smoke_test_variables functional_test_variables ; do
+      export VARIANT
+      # for each field ending in _password whose equivalent _email-ending field has the value $ACCOUNT_EMAIL
+      for password_field in  $(./sops-wrapper -d jenkins-vars/jenkins.yaml|yq -crM '.[env.VARIANT][env.STAGE_LOWER] as $vars|$vars|to_entries|.[]|select((.key|endswith("_email")) and .value == env.ACCOUNT_EMAIL).key|sub("_email$"; "_password")|select($vars[.])') ; do
+        ./sops-wrapper --set "[\"${VARIANT}\"][\"${STAGE_LOWER}\"][\"${password_field}\"] \"${new_password}\"" jenkins-vars/jenkins.yaml
+      done
+    done
+  done
+
+  commit_and_create_github_pr "Change functional test DMP account passwords" "## Summary\r\nChanges all functional test/smoke test/smoulder test DMP account passwords for ${STAGE}."
```

Comment:
A thought: if we want to reduce the amount of time between merging the new creds and syncing them, should we include this in the PR description? "Be ready to do the next step of the pipeline (syncing) when you merge this PR"

Reply:
Quite possibly yes